apps/meteor/client/views/marketplace/hooks/useFilteredApps.ts

```diff
@@ -49,70 +50,53 @@ export const useFilteredApps = ({
+			az: (apps: App[]) => apps.sort((firstApp, secondApp) => sortAppsByAlphabeticalOrInverseOrder(firstApp.name, secondApp.name)),
+			za: (apps: App[]) => apps.sort((firstApp, secondApp) => sortAppsByAlphabeticalOrInverseOrder(secondApp.name, firstApp.name)),
+			mru: (apps: App[]) =>
+				apps.sort((firstApp, secondApp) => sortAppsByClosestOrFarthestModificationDate(firstApp.modifiedAt, secondApp.modifiedAt)),
+			lru: (apps: App[]) =>
+				apps.sort((firstApp, secondApp) => sortAppsByClosestOrFarthestModificationDate(secondApp.modifiedAt, firstApp.modifiedAt)),
 		};
 
-		if (context && context === 'enterprise') {
-			filtered = apps.filter(({ categories }) => categories.includes('Enterprise'));
-		}
-
-		if (sortingMethod) {
-			filtered = sortingMethods[sortingMethod]();
-		}
-
-		const filterByPurchaseType: Record<string, () => App[]> = {
-			paid: () => filtered.filter(filterAppsByPaid),
-			enterprise: () => filtered.filter(filterAppsByEnterprise),
-			free: () => filtered.filter(filterAppsByFree),
+		const filterByPurchaseType: Record<string, (apps: App[]) => App[]> = {
+			all: fallback,
+			paid: (apps: App[]) => apps.filter(filterAppsByPaid),
+			enterprise: (apps: App[]) => apps.filter(filterAppsByEnterprise),
+			free: (apps: App[]) => apps.filter(filterAppsByFree),
 		};
 
-		if (purchaseType && purchaseType !== 'all') {
-			filtered = filterByPurchaseType[purchaseType]();
-
-			if (!filtered.length) shouldShowSearchText = false;
-		}
-
-		if (status && status !== 'all') {
-			filtered = status === 'enabled' ? filtered.filter(filterAppsByEnabled) : filtered.filter(filterAppsByDisabled);
-
-			if (!filtered.length) shouldShowSearchText = false;
-		}
-
-		if (Boolean(categories.length) && Boolean(text)) {
-			filtered = filtered.filter((app) => filterAppsByCategories(app, categories)).filter(({ name }) => filterAppsByText(name, text));
-			shouldShowSearchText = true;
-		}
+		const filterByStatus: Record<string, (apps: App[]) => App[]> = {
+			all: fallback,
+			enabled: (apps: App[]) => apps.filter(filterAppsByEnabled),
+			disabled: (apps: App[]) => apps.filter(filterAppsByDisabled),
+		};
 
-		if (Boolean(categories.length) && !text) {
-			filtered = filtered.filter((app) => filterAppsByCategories(app, categories));
-			shouldShowSearchText = false;
-		}
+		const filterByContext: Record<string, (apps: App[]) => App[]> = {
+			explore: fallback,
+			installed: fallback,
+			private: fallback,
+			enterprise: (apps: App[]) => apps.filter(({ categories }) => categories.includes('Enterprise')),
+			requested: (apps: App[]) => apps.filter(({ appRequestStats, installed }) => Boolean(appRequestStats) && !installed),
+		};
 
-		if (!categories.length && Boolean(text)) {
-			filtered = filtered.filter(({ name }) => filterAppsByText(name, text));
-			shouldShowSearchText = true;
-		}
+		const filtered = compose(
```

Comment:
use this: https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array/ReduceRight